lib/components/providers/theme/themeProvider.tsx

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState, PropsWithChildren } from "react";
+
+import { ThemeContext, initialThemeState } from "./themeContext";
+
+export const ThemeProvider = ({ children }: PropsWithChildren) => {
+    const savedThemeLocal = window?.localStorage.getItem("globalTheme");
+    const [theme, setTheme] = useState(
+        savedThemeLocal ?? initialThemeState.theme,
+    );
+
+    useEffect(() => {
+        if (savedThemeLocal) {
+            setTheme(savedThemeLocal);
+        }
+    }, []);
+
+    useEffect(() => {
+        window?.localStorage.setItem("globalTheme", theme);
+        document.body.classList.add(`theme--${theme}`);
+        document.body.style.backgroundColor =
+            "var(--semantic-color-background-secondary-base)";
```

Comment:
Can we add a checker if the `theme--${theme}` does not exist then only add `document.body.classList.add(`theme--${theme}`);`